tap_google_sheets/transform.py

```diff
@@ -49,42 +49,53 @@ def transform_file_metadata(file_metadata):
 
 # Convert Excel Date Serial Number (excel_date_sn) to datetime string
 # timezone_str: defaults to UTC (which we assume is the timezone for ALL datetimes)
-def excel_to_dttm_str(excel_date_sn, timezone_str=None):
+def excel_to_dttm_str(string_value, excel_date_sn, timezone_str=None):
     if not timezone_str:
         timezone_str = 'UTC'
+    is_error = False
     tzn = pytz.timezone(timezone_str)
     sec_per_day = 86400
     excel_epoch = 25569 # 1970-01-01T00:00:00Z, Lotus Notes Serial Number for Epoch Start Date
     epoch_sec = math.floor((excel_date_sn - excel_epoch) * sec_per_day)
     epoch_dttm = datetime(1970, 1, 1)
-    excel_dttm = epoch_dttm + timedelta(seconds=epoch_sec)
+    # For out of range values, it will throw OverflowError and it would return the string value
+    # as passed in the sheets without any conversion
+    try:
+        excel_dttm = epoch_dttm + timedelta(seconds=epoch_sec)
+    except OverflowError:
+        is_error = True
+        return str(string_value), is_error
     utc_dttm = tzn.localize(excel_dttm).astimezone(pytz.utc)
     utc_dttm_str = strftime(utc_dttm)
-    return utc_dttm_str
+    return utc_dttm_str, is_error
```

Comment:
We can pass False directly here and can remove the is_error variable.

Reply:
Updated the code